views/calendar/calendar.js

```diff
@@ -0,0 +1,416 @@
+                    // Log the Birth Event
+                    if (getMonth(x[i].BirthDate) != undefined) {
+                        pushEvents(x[i], x[i].BirthDate, " was born", "", "");
+                    }
+                    // Log the Death Event
+                    if (getMonth(x[i].DeathDate) != undefined) {
+                        pushEvents(x[i], x[i].DeathDate, " died", "", "");
+                    }
+                    // Log Marriage Events
+                    if (x[i].Spouses.length != 0) {
+                        var spouseIDs = Object.keys(x[i].Spouses);
+                        for (var j = 0; j < spouseIDs.length; j++) {
+                            if (getMonth(x[i].Spouses[spouseIDs[j]].marriage_date) != undefined) {
+                                pushEvents(
+                                    x[i],
+                                    x[i].Spouses[spouseIDs[j]].marriage_date,
+                                    " marriage to ",
+                                    x[i].Spouses[spouseIDs[j]].Name,
+                                    x[i].Spouses[spouseIDs[j]].ShortName
+                                );
+                            }
+                        }
+                        //if (getMonth(x[i].Spouses[i].marriage_date) != undefined) {
+                        //spanushEvents(x[i], x[i].Spouses.marriage_date, "  married ", );
+                    }
+                    if (i == x.length - 1) {
+                        // Hide Loader when finished iterating through profile
+                        $(".loader").hide();
+                        $(".calendar").css("display", "block");
+                        $(".calendarMenu").css("display", "block");
+                        // Open the Current Month
+                        $(`#${months[new Date().getMonth()]}`)
+                            .addClass("active box rounded orange")
+                            .css("display", "block");
+                    }
+                }
+                // Now we can sort our content for the calendar, by day and then by year.
+                for (var i = 0; i < 12; i++) {
+                    var array = Object.keys(AllMonths)[i];
+                    AllMonths[array].sort((a, b) => {
+                        if (a.day === b.day) {
+                            return a.year < b.year ? -1 : 1;
+                        } else {
+                            return a.day < b.day ? -1 : 1;
+                        }
+                    });
+                }
+                // Then we insert the content into the correct month
+                // This could be simplified...
+                for (var i = 0; i < x.length; i++) {
+                    try {
+                        $(`#Jan`).append(`
+                        <span>${AllMonths.Jan[i].day} ${getMonth(AllMonths.Jan[i].date)}
+                        ${AllMonths.Jan[i].year} <a href="https://www.wikitree.com/wiki/${AllMonths.Jan[i].id}">
```

Comment:
Those looks the same, here would the map function on array also work, just apply own function, that receives month name as parameter and produces that span. (or appends it directly, without returning)